skimage/filters/_gaussian.py

```diff
@@ -155,3 +155,117 @@ def _guess_spatial_dimensions(image):
+        used as the standard deviation value for all axes.
+    sigma2 : scalar or sequence of scalars, optional (default is None)
+        Standard deviation(s) for the Gaussian kernel with the larger sigmas
+        across all axes. The standard deviations are given for each axis as a
+        sequence, or as a single number, in which case the single number is
+        used as the standard deviation value for all axes. If None is given
+        (default), sigmas for all axes are calculated as 1.6 * sigma1.
+    mode : {'reflect', 'constant', 'nearest', 'mirror', 'wrap'}, optional
+        The ``mode`` parameter determines how the array borders are
+        handled, where ``cval`` is the value when mode is equal to
+        'constant'. Default is 'nearest'.
+    cval : scalar, optional
+        Value to fill past edges of input if ``mode`` is 'constant'. Default
+        is 0.0
+    multichannel : bool, optional (default: False)
+        Whether the last axis of the image is to be interpreted as multiple
+        channels. If True, each channel is filtered separately (channels are
+        not mixed together).
+    truncate : float, optional (default is 4.0)
+        Truncate the filter at this many standard deviations.
+
+    Returns
+    -------
+    filtered_image : ndarray
+        the filtered array
+
+    Notes
+    -----
+    This function will subtract an array filtered with a guassian kernel
+    with sigmas given by ``sigma2`` from an array filtered with a gaussian
+    kernel with sigmas provided by ``sigma1``. The values for ``sigma2`` must
+    always be greater than or equal to the corresponding values in ``sigma1``,
+    or a ``ValueError`` will be raised.
+
+    When ``sigma2`` is none, the values for ``sigma2`` will be calculated
+    as 1.6x the corresponding values in ``sigma1``. This approximates the
+    inverted Laplacian of Guassian, commonly used in edge and blob detection.
+
+    Input image is converted according to the conventions of ``img_as_float``.
+
+    Except for sigma values, all parameters are used for both filters.
+
+    Examples
+    --------
+    Apply a simple Difference of Gaussians filter to a color image:
+
+    >>> from skimage.data import astronaut
+    >>> from skimage.filters import difference_of_gaussians
+    >>> image = astronaut()
+    >>> filtered_image = difference_of_gaussians(image, 2, 10,
+    ...                                          multichannel=True)
+
+    Apply a Laplacian of Gaussian filter as approximated by the Difference
+    of Gaussians filter:
+
+    >>> filtered_image = difference_of_gaussians(image, 2, multichannel=True)
+
+    Apply a Difference of Gaussians filter to a grayscale image using different
+    sigma values for each axis:
+
+    >>> from skimage.data import camera
+    >>> image = camera()
+    >>> filtered_image = difference_of_gaussians(image, (2,5), (3,20))
+
+    """
+    image = img_as_float(image)
+    sigma1 = np.array(sigma1, dtype='float', ndmin=1)
```

Comment:
fascinating! Never knew about ndmin. =D

Reply:
yes I taught myself that for this PR, when I learned that giving `np.array` a scalar without forcing a dimension on it returns, unsurprisingly, a scalar. Since a numpy scalar has no length it ruined the numpy-based approach I had for handling my sigma inputs. There was another way I had of doing this using list comprehensions and checking for iterables, but that ended up looking way more complicated (to me) than what I went with.